src/components/Sidebar.vue

```diff
@@ -1,59 +1,78 @@
-
 <template>
-  <aside
-    class="w-56  inset-y-0  left-0 transform -translate-x-full transition duration-450 ease-in-out h-screen fixed overflow-y-auto"
-    :class="{' -translate-x-0':showSideBar}" aria-label="Sidebar">
-    <div class="overflow-y-auto py-4 px-3  bg-gray-100 h-screen dark:bg-gray-800">
-      <div
-        class="flex items-center justify-between p-2 tracking-widest text-base font-bold text-gray-900 rounded-lg dark:text-white ">
-        <p class="p-[10px]">
-          BeforeClass
-        </p>
-        <Button @click="showSideBar = !showSideBar">
-          <img
-            class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white "
-            src="../assets/arrowLeft.svg" alt="">
+    <div class="bg-gray-100 dark:bg-gray-800 p-5 pt-8 relative h-screen min-w-[14rem] duration-300 w-56"
+        :class="{'w-20 min-w-[5rem]':showSideBar}">
+        <Button @click="showSideBar = !showSideBar" class=" text-gray-500  absolute right-5 top-9 dark:text-gray-400"
+            :class="{'rotate-180':showSideBar}">
+            <img class="w-6 h-6" src="../assets/arrowLeft.svg" alt="">
         </Button>
-      </div>
-      <ul class="space-y-2 p-[10px] mt-[15px]">
-        <ListItem list-image="timetable.svg" list-name='Timetable' link-href="#" />
-        <ListItem list-image="modules.svg" list-name='Modules' link-href="#" />
+        <div class="inline-flex">
+            <h1 class="block float-left font-bold tracking-widest p-[10px] dark:text-white"
+                :class="{'scale-0':showSideBar}">
+                BeforeClass
+            </h1>
+        </div>
+        <ul class="space-y-2 p-[10px]  mt-[15px]" :class="{'px-0':showSideBar}">
+            <li>
```

Comment:
should use the `SidebarListItem` component here

Reply:
Not using SidebarListItem because Im using the showSideBar state to hide and show the list name. 
My thought process was that for such a specific use case (side bar list) and for only a few lists, might not need to create a separate component just for this, having to pass the state as a prop might make the SidebarListItem component more complex for a simple use case.
I was thinking we should try to keep  things simple and not over-complicate when not needed, thoughts?

Reply:
that makes sense, let's keep it simple, thanks for raising this! resolving this comment